app/src/utils/DataFetcher.ts

```diff
@@ -3,49 +3,80 @@
 // establish the API types.
 
 import { MetadataMap, Row } from "./DatasetTypes";
+import { diabetes } from "./FakeData";
 import FakeMetadataMap from "./FakeMetadataMap";
+import { DataFrame } from "data-forge";
+import { reshapeColsToRows } from "./datasetutils";
 
-class DataFetcher {
-  private async loadData(url: string): Promise<Response> {
-    const r = await fetch(url);
-    return await r.json();
-  }
+const colMap = {
+  DP05_0070E: "All Races",
+  DP05_0071E: "Hispanic",
+  DP05_0077E: "White, Non-Hispanic",
+  DP05_0078E: "Black, Non-Hispanic",
+  DP05_0080E: "Asian, Non-Hispanic",
+  DP05_0079E: "American Indian/Alaskan Native, Non-Hispanic",
+};
 
-  private getUrl(datasetId: string): string {
-    if (datasetId === "state_names") {
-      return "https://api.census.gov/data/2010/dec/sf1?get=NAME&for=state:*";
-    } else if (datasetId === "county_names") {
-      return "https://api.census.gov/data/2010/dec/sf1?get=NAME&for=county:*&in=state:*";
-    } else if (datasetId === "pop_by_race") {
-      return (
-        "https://api.census.gov/data/2018/acs/acs5/profile" +
-        "?get=NAME,DP05_0070E,DP05_0071E,DP05_0077E,DP05_0078E,DP05_0080E" +
-        "&for=county:*&in=state:*"
-      );
-    }
-    throw new Error("Unknown dataset");
-  }
+const colsToMerge = [
+  "DP05_0081E", // Native Hawaiian and Other Pacific Islander alone, Non-Hispanic
+  "DP05_0082E", // Some other race alone, Non-Hispanic
+  "DP05_0083E", // Two or more races, Non-Hispanic
+];
 
-  private convertJson(data: any): Row[] {
-    let realData = data;
-    const headers = realData[0];
-    realData = realData.slice(1);
-    realData = realData.map((row: Array<any>) => {
-      const newRow: { [key: string]: any } = {};
-      for (let i = 0; i < headers.length; i++) {
-        newRow[headers[i]] = row[i];
-      }
-      return newRow;
+async function getAcsStatePopulations() {
+  const populationCols = colsToMerge.concat(Object.keys(colMap));
+  const params = ["NAME"].concat(populationCols).join(",");
+  let r = await fetch(
+    `https://api.census.gov/data/2018/acs/acs5/profile?get=${params}&for=state:*`
+  );
+  let json = await r.json();
+  let df = new DataFrame({ columnNames: json[0], rows: json.slice(1) })
+    .parseInts(populationCols)
+    .generateSeries({
+      "Other race, Non-Hispanic": (row) =>
+        colsToMerge.map((c) => row[c]).reduce((acc, v) => acc + v),
+    })
+    .dropSeries(colsToMerge)
+    .renameSeries(colMap)
+    .renameSeries({
+      state: "state_fips_code",
+      NAME: "state_name",
     });
-    return realData;
-  }
+  const renamedPopCols = ["Other race, Non-Hispanic"].concat(
+    Object.values(colMap)
+  );
+  return reshapeColsToRows(df, renamedPopCols, "population", "race");
+}
 
+function getDiabetesFrame() {
+  return new DataFrame(diabetes)
+    .dropSeries([
+      "DIABETES_NO_REFUSED",
+      "PREDIABETES_YES_YESPREGNANT",
+      "PREDIABETES_NO_UNSURE_REFUSED",
+      "COPD_YES",
```

Comment:
Actually could we keep COPD_YES for now? I was using diabetes and COPD as two different options for MadLib options to demonstrate how the report updates. It's also on prioritized variable list.

Although... does product specifically want them to be multiple datasets (brfss_diabetes and brfss_copd) ? That seems more aligned with how I think they want the data download catalog...

Do you have more insight here? Are we sure if product wants brfss_diabetes and brfss_copd as separate datasets? If so, I can add COPD myself after you submit.

Reply:
Sure thing, added it back. I don't know if we'll want it as two datasets or not, I think it's fine to keep it in one for now unless/until we encounter issues or hear otherwise